package/src/lib/components/attach-step/AttachStep.component.tsx

```diff
@@ -57,12 +58,12 @@ export function AttachStep<T>({ children, fill = false, index }: AttachStepProps
   const childRef = useRef<View>(null);
 
   useEffect(() => {
-    if (current === index) {
+    if (checkIndex(index, current)) {
       childRef.current?.measureInWindow((x, y, width, height) => {
         changeSpot({ height, width, x, y });
       });
     }
-  }, [changeSpot, current, index]);
+  }, [changeSpot, current, index.toString()]);
```

Comment:
This is needed to compare arrays, otherwise, they will always differ and `useEffect` will be called infinite times.